src/legacy/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable.ts

```diff
@@ -159,17 +191,16 @@ export class VisualizeEmbeddable extends Embeddable {
    * @param {Element} domNode
    * @param {ContainerState} containerState
    */
-  public render(domNode: HTMLElement, containerState: ContainerState) {
-    this.panelTitle = this.getPanelTitle(containerState);
-    this.timeRange = containerState.timeRange;
-    this.query = containerState.query;
-    this.filters = containerState.filters;
+  public render(domNode: HTMLElement) {
+    this.timeRange = _.cloneDeep(this.input.timeRange);
```

Comment:
`this.timeRange` is used above to check if the previous timeRange passed in has changed.  Could possibly be a more first class way of knowing what exactly changed when a new input is passed in...

and yea, deeply cloned because somewhere something was mutating them and then messing up the comparison of "did something change, should I update my observers?".  Kind of like redux tree, input should never be mutated directly.  I couldn't figure out where though!